pkg/command/root.go

```diff
@@ -282,6 +261,55 @@ func ensureCLIInstanceID() (string, error) {
 	return cliID, nil
 }
 
+func shouldSkipTelemetry(cmd *cobra.Command) bool {
+	skipTelemetryCommands := []string{
+		// The shell completion setup is not interactive, so it should not trigger a prompt
+		"tanzu __complete",
+		"tanzu completion",
+		// Common first command to run,
+		"tanzu version",
+		// should skip telemetry for "telemetry" plugin
+		"tanzu telemetry",
+	}
+	skipTelemetry := false
+	for _, cmdPath := range skipTelemetryCommands {
+		if strings.HasPrefix(cmd.CommandPath(), cmdPath) {
+			skipTelemetry = true
+			break
+		}
+	}
+	return skipTelemetry
+}
+
+func shouldSkipPrompts(cmd *cobra.Command) bool {
+	// Prompt user for EULA and CEIP agreement if necessary, except for
+	skipCommands := []string{
+		// The shell completion setup is not interactive, so it should not trigger a prompt
+		"tanzu __complete",
+		"tanzu completion",
+		// Common first command to run,
+		"tanzu version",
+		// It would be a chicken and egg issue if user tries to set CEIP configuration
+		// using "tanzu config set env.TANZU_CLI_CEIP_OPT_IN_PROMPT_ANSWER yes"
+		"tanzu config set",
+		// Auto prompting when running these commands is confusing
+		"tanzu config eula",
+		"tanzu ceip-participation set",
+		// This command is being invoked by the kubectl exec binary where the user doesn't
+		// get to see the prompts and the kubectl command execution just gets stuck, and it
+		// is very hard for users to figure out what is going wrong
+		"tanzu pinniped-auth",
+	}
+	skipPrompts := false
```

Comment:
We can have a separate method with common code in `shouldSkipTelemetry` and `shouldSkipPrompts` methods.

Reply:
Updated